[Gabriel Waikin Loh Matienzo] iP

GabrielWLM opened the pull request.

Commit:
Add Gradle support

GabrielWLM committed:
Level-1

GabrielWLM committed:
Level-2

GabrielWLM committed:
Level-2

GabrielWLM committed:
Level-3

GabrielWLM committed:
Level-4

GabrielWLM committed:
Level-5

GabrielWLM committed:
Level-6

GabrielWLM committed:
branch-Level-7

GabrielWLM committed:
branch-Level-8

GabrielWLM committed:
Merge branch 'branch-Level-7'

GabrielWLM committed:
Merge branch 'branch-Level-8'

GabrielWLM committed:
A-MoreOOP

GabrielWLM committed:
A-Packages

GabrielWLM committed:
A-JUnit

GabrielWLM committed:
A-Jar

GabrielWLM committed:
A-JavaDoc

GabrielWLM committed:
A-CodingStandard

GabrielWLM committed:
Level-9

GabrielWLM committed:
Merge branch 'branch-A-JavaDoc'

GabrielWLM committed:
Merge branch 'branch-A-CodingStandard'

GabrielWLM committed:
Merge branch 'branch-Level-9'

GabrielWLM commented:
### LUKE

> "Luke, I am your father"

**LUKE** helps **you** remember your tasks, events and deadlines for you!

- text-based
- easy to learn
- FAST _SUPER FAST_ to use
- All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 Managing tasks
 Managing deadlines

- [X] Managing tasks
- [X] Managing deadlines
- [ ] Reminders (coming soon)

Here's the `main` method

>public class Main {
>    public static void main(String[] args) {
>        Application.launch(MainApp.class, args);
>    }
>}

GabrielWLM committed:
A-Gradle

GabrielWLM committed:
Merge branch 'branch-A-Gradle'

GabrielWLM committed:
Level-10
git commit -m Level-10
git commit -m Level-10

GabrielWLM committed:
Level-10

GabrielWLM committed:
Merge branch 'branch-Level-10'

RachelCheah commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+
+        String response = scanner.nextLine();
+
+        while (!response.equals("bye")) {
```
I think you should try splitting the logic in this while loop into different files/methods

RachelCheah reviewed (commented):
The code seems to adhere to the coding standard specified overall!

aakanshanarain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+
+        ArrayList<String> list = new ArrayList<String>();
```
Variable 'list' could be more descriptive; i.e, describe the purpose of the variable

aakanshanarain commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                int taskNo = Integer.parseInt(response.replaceAll("\\D", "")) - 1;
+                System.out.println(lineBreak);
+                System.out.println("\t Nice! I've marked this task as done:");
+                list.set(taskNo, list.get(taskNo).substring(0,4) + "X" + list.get(taskNo).substring(5));
```
> substring(0,4) 
`substring(0, 4)`

aakanshanarain reviewed (commented):
Looks alright! Not many errors regarding the coding standards here!

jivesh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+    private static void save(ArrayList<String> tasks) throws IOException {
+        File f = new File("tasks.txt");
+        System.out.print(f.createNewFile());
+        FileWriter fw = new FileWriter("tasks.txt");
```
A longer, more descriptive name (like simply `fileWriter`) might be better here.

jivesh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                list.add(amount, "[T][ ] " + response.substring(5));
+                amount++;
+                save(list);
+            } else if (response.matches("deadline .+")) {
```
For response matching, you might benefit from using Enums :)

jivesh commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                System.out.println("☹ OOPS!!! The description of an event cannot be empty.");
+                System.out.println(lineBreak);
+            } else {
+                System.out.println(lineBreak);
```
You could consider abstracting the
```java
System.out.println(lineBreak);
System.out.println(something);
System.out.println(lineBreak);
```
to avoid repeated code

jivesh commented:
Good code, but you might benefit from splitting your logic into multiple files :)

GabrielWLM committed:
Update sample data

Sample data has many blank lines.

Blank lines are signs of messy code.

Let's delete the blank lines.

It is only way to get rid of the blank lines.

This is an example for A-FullCommitMessage.

GabrielWLM committed:
A-Assertions

GabrielWLM committed:
A-CodeQuality

GabrielWLM committed:
Merge pull request #1 from GabrielWLM/branch-A-Assertions

A-Assertions

GabrielWLM committed:
Merge branch 'master' into branch-A-CodeQuality

GabrielWLM committed:
Merge pull request #2 from GabrielWLM/branch-A-CodeQuality

A-CodeQuality

GabrielWLM committed:
Merge pull request #4 from GabrielWLM/add-gradle-support

Add Gradle support

GabrielWLM committed:
Refactor duke

GabrielWLM committed:
import junit dependencies and improve code quality

GabrielWLM committed:
add README and Ui screenshot

GabrielWLM committed:
Set theme jekyll-theme-modernist

GabrielWLM committed:
fix folder creation and jar execution

GabrielWLM committed:
Merge branch 'master' of https://github.com/GabrielWLM/ip
